### ISSUE TITLE
Can't use ApiFactory.csmsOcppServer function

### ISSUE THREAD

**Issue by zafar-apk** [IZIVIA/ocpp-toolkit#54]
I added your latest release using gradle kts:

 implementation("com.izivia:ocpp-toolkit:1.2.2")

It successfully added. But i can't really use it as the toolkit uses other modules using implementation, not api. So it is not transitive. 

As a result i can't access modules: 
:generic-api
:ocpp-transport
:operation-information

But files and classes are required to use ApiFactory.csmsOcppServer function
Here is the screenshot of the errors i am getting.
![Screenshot_2024-01-14_19-36-05](https://github.com/IZIVIA/ocpp-toolkit/assets/78132342/df78d916-f0d9-4048-9897-eb2c54daa74e)

**Comment by zafar-apk** [comment]
There are no samples of csmsOcppServer but in the readme you specified that it can be also used for implementing csms system. But anyway thank you guys for sharing this project. So helpfull

**Comment by edwardmp** [comment]
Noticed the same. It seems you just need to also include those modules in your gradle build config.

I do believe this can be fixed at the root, bij changing the `implementation` expressions to `api` expressions in the toolkit module gradle config so that these classes are exposed instead of hidden.

@lilgallon would you agree with that?

**Comment by lilgallon** [comment]
Yes you are right, as of now, you need to add:

```gradle
implementation("com.izivia:generic-api:1.2.2")
implementation("com.izivia:ocpp-transport:1.2.2")
implementation("com.izivia:operation-information:1.2.2")
```

in your project.

In our codebase, we should be using `api` for these modules. I’m not sure why that hasn’t been the case so far. Perhaps it hasn’t caused any issues because it hasn’t been widely utilized.

It would be fantastic to attract new users to the library. It’s already successfully managing thousands of charging stations from various manufacturers in production. However, no dedicated human resources have been allocated to maintain or promote this repository, that is why it looks "dead" whereas it is not

**Comment by edwardmp** [comment]
@lilgallon library looks great! Keen to start using it, and giving back. Would you be open to a PR that changes to `api`?

**Comment by lilgallon** [comment]
Yes absolutely, pull requests are welcome. There are no contribution guidelines for now, but as long as the code is clean, it should be good :)

**Comment by edwardmp** [comment]
Oh and one slightly offtopic question: how does one use the generic API? I dove into the code and got the impression there is simply no way to use it instead of having OCPP version specific handlers (e.g. when using `ApiFactory.csmsOcppServer`), but perhaps I might be mistaken

**Comment by lilgallon** [comment]
I am not the main maintainer of the project, so I may be wrong. `ApiFactory.csmsOcppServer` initializes and starts a CSMS server using the callbacks you pass through `csmsApiCallbacks`. It uses ocpp16 and ocpp20 implementations, so you must have the required dependencies in your codebase.

That is a way to use it. Another way would be to simply use the parser to parse the incoming messages and then process the messages according to their type. When using SOAP, you can have two endpoints to differentiate OCPP versions (`/api/ocpp16`, `/api/ocpp20` let's say), and when using json (websocket), you can use the transport implementation of the lib and implement `OcppWampServerHandler` (you have the ocpp version in `WampMessageMeta`). The ws transport is used in production that way

**Comment by edwardmp** [comment]
Thanks for the elaborate response @lilgallon.

What I mean is that it seems the generic implementation can't actually be used; 

For instance there is this method to get the CS generic API: https://github.com/IZIVIA/ocpp-toolkit/blob/bd8e7334ae05ea75d02d96a508269acbe076bcd8/toolkit/src/main/kotlin/com/izivia/ocpp/integration/CSMS.kt#L61

Yet, in the same class there is literally no way to ever set the generic API implementation class to be used (see https://github.com/IZIVIA/ocpp-toolkit/blob/bd8e7334ae05ea75d02d96a508269acbe076bcd8/toolkit/src/main/kotlin/com/izivia/ocpp/integration/CSMS.kt#L36)

So I really wonder how the generic API can be used. Seems right now it's impossible to use
